service/storage_service.cc

```diff
@@ -1488,9 +1488,12 @@ future<> storage_service::stop() {
     // make sure nobody uses the semaphore
     node_ops_singal_abort(std::nullopt);
     _listeners.clear();
-    return _schema_version_publisher.join().finally([this] {
-        return std::move(_node_ops_abort_thread);
-    });
+    try {
+        co_await _schema_version_publisher.join();
+    } catch (...) {
+        slogger.error("schema_version_publisher failed: {}", std::current_exception());
+    }
+    co_await std::move(_node_ops_abort_thread);
```

Comment:
I'm not an expert in coroutines but shouldn't this be return or co_return instead of co_await?

Reply:
co_await = wait for this future
co_return value = return this value
co_return = exist the function now

We want to wait for the thread to complete, so we call co_await.